### PR TITLE
Improve error message when --destination exists

### PR DESCRIPTION
**Pull request by ewels**
Add in a new custom error message when `--directory` exists, so that it's clearer what is wrong.

Note that these should probably be raising exceptions rather than doing `sys.exit`.. But I'm a bit behind what the current methodology is so I'll leave that for a suggestion / separate PR.

Before submitting a PR to the `dev` branch:
- [x] Tests passing
- [x] Black formatting
- [x] Rebase/merge the `dev` branch
- [x] Note in the CHANGELOG

Additional checks before submitting a PR to the `master` branch:
- Change version in `setup.py` (?)

### CODE REVIEW

File: CHANGELOG.md [hunk 1]
```diff
@@ -96,9 +114,10 @@ Please add a _short_ line describing the PR you make, if the PR implements a spe
 * Add a `--token-path` argument to tell where the token should be saved and which token to be used. ([#329](https://github.com/ScilifelabDataCentre/dds_cli/pull/329))
 * Remove `--username` option ([#331](https://github.com/ScilifelabDataCentre/dds_cli/pull/331))
 * Add support for the zero-conf environment in dds_web ([#337](https://github.com/ScilifelabDataCentre/dds_cli/pull/337))
-* Increase request timeout to 30 ([#344]https://github.com/ScilifelabDataCentre/dds_cli/pull/344)
+* Increase request timeout to 30 ([#344]<https://github.com/ScilifelabDataCentre/dds_cli/pull/344>)
```
Comment (inaod568) [inline CHANGELOG.md:117]: Did you mean to put `<>` and not `()`?

**Review by inaod568 — APPROVED** [review]
💪🏻

File: CHANGELOG.md (hunk 1 above)
Comment (inaod568) [reply CHANGELOG.md:117]:
```suggestion
* Increase request timeout to 30 ([#344](https://github.com/ScilifelabDataCentre/dds_cli/pull/344))
```

Comment (ewels) [reply CHANGELOG.md]: Nope, my editor automatically fixes markdown so I bet that I hit save before I finished writing the markdown and it surrounded the URL for me..

Comment (ewels) [reply CHANGELOG.md]: (naked URLs should have < url > for properly valid markdown)

Comment (ewels) [reply CHANGELOG.md]: Ah it's not my changelog - ok then yeah it was broken markdown before (no brackets) and it automatically fixed it in a dumb way.. 😆

**Pull request merged**